packages/fxa-auth-server/scripts/apple-transfer-users/apple-transfer-users.js

```diff
@@ -0,0 +1,262 @@
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+'use strict';
+
+// This script is used to import users from Apple into FxA. It consumes a CSV
+// containing the Apple `transfer_sub` id and exchanges it for their profile information.
+// It then creates/updates the user and links the Apple account to the FxA account.
+//
+// Usage: node scripts/apple-transfer-users.js -i <input file> -o <output file>
+
+const fs = require('fs');
+const path = require('path');
+const program = require('commander');
+
+const axios = require('axios');
+const random = require('../../lib/crypto/random');
+const uuid = require('uuid');
+
+const log = require('../../lib/log')({});
+const config = require('../../config').config.getProperties();
+const Token = require('../../lib/tokens')(log, config);
+const AuthDB = require('../../lib/db')(config, log, Token);
+
+const GRANT_TYPE = 'client_credentials';
+const SCOPE = 'user.migration';
+const USER_MIGRATION_ENDPOINT = 'https://appleid.apple.com/auth/usermigrationinfo';
+
+const APPLE_PROVIDER = 2;
+
+export class AppleUser {
+  constructor(email, transferSub, uid, alternateEmails, db) {
+    this.email = email;
+    this.transferSub = transferSub;
+    this.uid = uid;
+    this.alternateEmails = alternateEmails || [];
+    this.db = db;
+  }
+
+  // Exchanges the Apple `transfer_sub` for the user's profile information and
+  // moves the user to the new team.
+  // Ref: https://developer.apple.com/documentation/sign_in_with_apple/bringing_new_apps_and_users_into_your_team#3559300
+  async exchangeIdentifiers(accessToken) {
+    try {
+      const options = {
+        transfer_sub: this.transferSub,
+        client_id: config.appleAuthConfig.clientId,
+        client_secret: config.appleAuthConfig.clientSecret,
+      };
+      const res = await axios.post(USER_MIGRATION_ENDPOINT,
+        new URLSearchParams(options).toString(),
+        {
+          headers: {
+            Authorization: `Bearer ${accessToken}`,
+          },
+        },
+      );
+
+      // Data here contains `sub`, `email` and `is_private_email`
+      this.appleUserInfo = res.data;
+      return res.data;
+    } catch (err) {
+      this.setFailure(err);
+      if (err.response && err.response.status === 429) {
+        console.error(`Rate limit exceeded, try again later: ${this.transferSub}`);
+      } else {
+        console.error(`Something went wrong with transfer: ${this.transferSub} ${err}`);
+      }
+    }
+  }
+
+  setSuccess(accountRecord) {
+    this.success = true;
+    this.accountRecord = accountRecord;
+  }
+
+  setFailure(err) {
+    this.success = false;
+    this.err = err;
+  }
+
+  async createLinkedAccount(accountRecord, sub) {
+    // If the user already has a linked account, delete it and create a new one.
+    await this.db.deleteLinkedAccount(accountRecord.uid, APPLE_PROVIDER);
```

Comment:
There shouldn't be any harm in deleting a linked third party auth account. The feature is broadcasted so no real users should have this set.